x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx

```diff
@@ -375,6 +369,12 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
   }, [ruleType, previousRuleType, getFields]);
 
   usePersistentAlertSuppressionState({ form });
+  usePersistentNewTermsState({
+    form,
+    ruleTypePath: 'ruleType',
+    newTermsFieldsPath: 'newTermsFields',
```

Comment:
Let's use constants for `newTermsFields` and `historyWindowSize` (`ruleType` is addressed in https://github.com/elastic/kibana/pull/199887) defined inside corresponding edit component folders. `usePersistentNewTermsState()` should import those constants under the hood. Checkout a comment from Vitalii https://github.com/elastic/kibana/pull/199887#discussion_r1848174619.